console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import {
+  Select,
+  SelectVariant,
+  SelectOption,
+  SelectOptionObject
+} from "@patternfly/react-core";
+import {
+  TypeAheadMessage,
+  TYPEAHEAD_REQUIRED_LENGTH
+} from "constants/constants";
+import { ISelectOption } from "utils";
+
+interface ITypeAheadSelectProps {
```

Comment:
an you try to use SelectProps to get all the properties, We should avoid to create unnecessary properties those already are available in importing component interface. here you can extend SelectProps and avoid the properties declare those are already threre.
interface ITypeAheadSelectProps extends SelectProps {
//declare properties which are not available in SelectProps
}

we will follow this practice for all the components.